File: monkey/token/token.go
package token

type TokenType string
type Token struct {
	Type    TokenType
	Literal string
}

const (
	ILLEGAL = "ILLEGAL" // 开始
	EOF     = "EOF"     // 结束

	// 标识符（变量、函数名等） 定义符（数值、字符串、日期等）
	IDENT  = "IDENT" // add, foobar, x, y,...
	INT    = "INT"   // 1343456
	STRING = "STRING"

	// 操作符
	ASSIGN   = "="
	PLUS     = "+"
	MINUS    = "-"
	BANG     = "!"
	ASTERISK = "*"
	SLASH    = "/"

	LT = "<"
	GT = ">"

	EQ     = "=="
	NOT_EQ = "!="

	// 分隔符
	COMMA     = ","
	SEMICOLON = ";"
	LPAREN    = "("
	RPAREN    = ")"
	LBRACE    = "{"
	RBRACE    = "}"
	COLON     = ":"

	LBRACKET = "["
	RBRACKET = "]"

	// 关键字
	FUNCTION = "FUNCTION"
	LET      = "LET"
	TRUE     = "TRUE"
	FALSE    = "FALSE"
	IF       = "IF"
	ELSE     = "ELSE"
	RETURN   = "RETURN"
)

// 定义关键字
var keywords = map[string]TokenType{
	"fn":     FUNCTION,
	"let":    LET,
	"true":   TRUE,
	"false":  FALSE,
	"if":     IF,
	"else":   ELSE,
	"return": RETURN,
}

// 检测当前字符是不是一个关键字
func LookupIdent(ident string) TokenType {
	if tok, ok := keywords[ident]; ok {
		return tok
	}
	return IDENT
}

File: lg-chinese/token.js
const TOKEN_TYPE = {
  ILLEGAL : "ILLEGAL",
	EOF     : "EOF" , 

	IDENT  : "IDENT" ,
	INT    : "INT"  , 
	STRING : "STRING",

  PRINT: 'PRINT',
  FUNCTION: 'FUNCTION',
  LET: 'LET',
   
  ASSIGN: "=",
  ASTERISK : "*",

  LPAREN  :  "(",
	RPAREN   : ")",
	LBRACE  :  "{",
	RBRACE   : "}",


  COMMA: ",",
	SEMICOLON: ";",
}

const NewToken = (tokenType, value) => {
  return {
    value,
    type: tokenType,
  }
}

module.exports = {
  NewToken,
  TOKEN_TYPE
}